interview-questions.md

```diff
@@ -8,33 +8,33 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
-Your answer:
+Your answer: JSX connects JS to HTML
 
-Researched answer:
+Researched answer: JSX stands for JavaScript Syntax Extension, which is an extension of JavaScript used in React, that will allow you to use HTML in JavaScript. you use className in JSX vs using class in HTML
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-Your answer:
+Your answer: Yarn is the command used to create a React app, th  json file is modified
 
-Researched answer:
+Researched answer: Yarn stands for Yet Another Resource Negotiator, and it is a package manager that was created by Facebook. Yarn will create a json paackage and a yarn.lock package is created as well when yarn is being ran
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: Anonymous functions do not have a name yet and are contained in ()
 
-Researched answer:
+Researched answer: These are functions that have not been declared yet, and can be used as an arguement for another function
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: state values and props in React are similar, they are both js objects. Props can get passed into a a component, and a state is inside of a component.
 
-Researched answer:
+Researched answer: Both props and state values are objects in JavaScript. Props are used to pass data from components and state values are controlled by the component itself and states are used to access props.
```

Comment:
State is a special variable in React and can be any data type, not just an object.